eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RoutingExtension.php

```diff
@@ -43,8 +69,77 @@ public function getName()
      *
      * @return \eZ\Publish\Core\MVC\Symfony\Routing\RouteReference
      */
-    public function getRouteReference($resource = null, $params = [])
+    public function getRouteReference($resource = null, $params = []): RouteReference
     {
         return $this->routeReferenceGenerator->generate($resource, $params);
     }
+
+    public function getPath($name, array $parameters = [], bool $relative = false): string
+    {
+        $referenceType = $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH;
+        if (is_object($name)) {
+            return $this->generateUrlForObject($name, $parameters, $referenceType);
+        }
+
+        return $this->urlGenerator->generate($name, $parameters, $referenceType);
+    }
+
+    public function getUrl($name, array $parameters = [], bool $schemeRelative = false): string
+    {
+        $referenceType = $schemeRelative ? UrlGeneratorInterface::NETWORK_PATH : UrlGeneratorInterface::ABSOLUTE_URL;
+        if (is_object($name)) {
+            return $this->generateUrlForObject($name, $parameters, $referenceType);
+        }
+
+        return $this->urlGenerator->generate($name, $parameters, $referenceType);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function generateUrlForObject(object $object, array $parameters, int $referenceType): string
+    {
+        $routeName = null;
+
+        if ($object instanceof Location) {
+            $routeName = UrlAliasRouter::URL_ALIAS_ROUTE_NAME;
+            $parameters += [
+                'locationId' => $object->id,
+            ];
+        } elseif ($object instanceof Content || $object instanceof ContentInfo) {
+            $routeName = UrlAliasRouter::URL_ALIAS_ROUTE_NAME;
+            $parameters += [
+                'contentId' => $object->id,
+            ];
+        } elseif ($object instanceof RouteReference) {
+            $routeName = $object->getRoute();
+            $parameters += $object->getParams();
+        } else {
+            throw new InvalidArgumentException('$object', 'Unsupported class: ' . get_class($object));
```

Comment:
It would be nice if this would not throw, to allow using these methods in other packages like Netgen Tags, which also used `path` and `url` to generate links to tag objects.

Reply:
That would be useful extension point, we will think of it. For now in all cases which are not supported by eZ Platform I've added fallback to replacement proposed by `symfony-cmf/routing`. 

// cc @bdunogier

Reply:
Thanks, I think this will be enough for now. I will test once you merge this.